### PR TITLE
Add check on cache length when purging entries

### PR DESCRIPTION
**Pull request by athum**
When the authenticator's cache is purged, there was a consistent `slice bounds out of range [:200] with capacity 181` panic occurring. Checking that the `count` variable here does not exceed the actual length of the cache should avoid this panic when the cache is purged.

### CODE REVIEW

File: digest.go [hunk 1]
```diff
@@ -80,7 +80,13 @@ func (da *DigestAuth) Purge(count int) {
 	}
 	cache := digestCache(entries)
 	sort.Sort(cache)
-	for _, client := range cache[:count] {
+
+	limit := count
+	if len(cache) < count {
+		limit = len(cache) - 1
+	}
```
Comment (abbot) [inline digest.go:87]: If I understand the issue correctly, when `Purge` is called with `count >= len(cache)`, this triggers a panic because of the range. I think an easier fix would be to just empty the entire cache in this case, right after grabbing a lock, because this is what I think this code effectively does. Or am I missing something?

Also, can you please add a test for this edge case?

Comment (athum) [reply digest.go:87]: @abbot sorry for the delayed response - you are correct! I think the current `count` parameter is to only clear the cache up to a certain point - is this something that can be removed in favor of clearing the entire cache?

**Comment by athum** [comment]
Moved to https://github.com/abbot/go-http-auth/pull/70.
